src/dtemplate.d

```diff
@@ -260,7 +260,7 @@ private bool match(RootObject o1, RootObject o2)
             printf("\te1 = %s '%s' %s\n", e1.type.toChars(), Token.toChars(e1.op), e1.toChars());
             printf("\te2 = %s '%s' %s\n", e2.type.toChars(), Token.toChars(e2.op), e2.toChars());
         }
-        if (!e1.equals(e2))
+        if (e1.type != e2.type || !e1.equals(e2))
```

Comment:
Thank you. Please add as a comment to the code. Additionally, comparing types is also done with .equals()